package/MDAnalysis/guesser/default_guesser.py

```diff
@@ -0,0 +1,606 @@
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the Lesser GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+
+r"""
+Default Guesser
+================
+.. _DefaultGuesser:
+
+DefaultGuesser is a generic guesser class that has basic guessing methods.
+This class is a general purpose guesser that can be used with most topologies,
+but being generic makes it the less accurate among all guessers.
+
+
+
+
+
+Classes
+-------
+
+.. autoclass:: DefaultGuesser
+   :members:
+   :inherited-members:
+
+"""
+from .base import GuesserBase
+import numpy as np
+import warnings
+import math
+
+import re
+
+from ..lib import distances
+from . import tables
+
+
+class DefaultGuesser(GuesserBase):
+    """
+    This guesser holds generic methods (not directed to specific contexts) for
+    guessing different topology attribute. It has the same methods which where
+    originally found in Topology.guesser.py. The attributes that can be
+    guessed by this class are:
+    * masses
+    * types
+    * elements
+    * angles
+    * dihedrals
+    * bonds
+    * improper dihedrals
+    * aromaticities
+
+    You can use this guesser either directly through an instance, or through
+    the :meth:`~MDAnalysis.core.universe.Universe.guess_TopologyAttrs` method.
+
+    Examples
+    --------
+    to guess bonds for a universe::
+
+        import MDAnalysis as mda
+        from MDAnalysisTests.datafiles import two_water_gro
+
+        u = mda.Universe(two_water_gro, context='default', to_guess=['bonds'])
+
+    .. versionadded:: 2.8.0
+
+    """
+    context = 'default'
+
+    def __init__(self, universe, **kwargs):
+        super().__init__(universe, **kwargs)
+        self._guesser_methods = {
+            'masses': self.guess_masses,
+            'types': self.guess_types,
+            'elements': self.guess_types,
+            'bonds': self.guess_bonds,
+            'angles': self.guess_angles,
+            'dihedrals': self.guess_dihedrals,
+            'impropers': self.guess_improper_dihedrals,
+            'aromaticities': self.guess_aromaticities,
+        }
+
+    def guess_masses(self, atom_types=None, indices_to_guess=None):
+        """Guess the mass of many atoms based upon their type.
+        For guessing masses through Universe.guess_TopologyAttrs():
```

Comment:
```suggestion
        For guessing masses through :meth:`~MDAnalysis.core.universe.Universe.guess_TopologyAttrs`:

```
I think this is the intended formatting?